### PR TITLE
feat(menu): facilitate use of input components in menu

### PR DESCRIPTION
**Pull request by nayfin**
use attribute directive md-menu-input-item on input components (e.g.
md-slider, md-input, etc) inside of menu to prevent menu from closing when
components are focused via click or tab. this will allow users to use menu
as a short form to input data without redirecting to another page or
calling a dialog box. Add styles to menu.scss to space and stretch components
properly. Add additional card to menu-demo to demonstrate use with all input
components.

closes issue 2612

### CODE REVIEW

**Comment by nayfin** [comment]
I signed it

**Comment by nayfin** [comment]
I signed it

**Comment by nayfin** [comment]
I fat fingered my email in `git config --global user.email`. Updated to correct email and restarted PR, but am still getting `cla: no` tag.

**Comment by DevVersion** [comment]
That seems to be because you have **2** commits pushed to this PR. And one of them is not signed with any correct `user.email`. I recommend squashing the commits together.

**Comment by nayfin** [comment]
@DevVersion I really appreciate the direction, but I still can't shake the `no: cla` label. I followed this [gist](https://gist.github.com/jbub/5766366). 

In bash I ran:
`git rebase --interactive HEAD~2`

Then vim opened with lines:
`pick <latest-commit> message`
`pick <previous-commit> message`

I added these two lines under those:
`pick <latest-commit> message`
`s <previous-commit> message`

I then fixed any merge conflicts ran `git add .` then `git rebase --continue` followed by `git push -f`

All checks have passed again, but I can't shake that `cla: no` label. Any advice would really be appreciated.

**Comment by nayfin** [comment]
I tried squashing the commits together, reverting the commit with wrong email, and editing the email. No luck. I'll close and reopen with a clean commit history. I love git but I always feel like I understand it just enough to shoot myself in the foot.

**Comment by jakpaw** [comment]
Did you open a new pull request? I would very much like to see it :)

**Comment by UltraMarco** [comment]
`[md-menu-in**t**put-item], [mdMenuInputItem], [mat-menu-in**t**put-item], [matMenuInputItem]`

File: src/lib/menu/menu-input-item.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import {Directive} from '@angular/core';
+
+
+/**
+ * Directive to allow users to use menu for user input by holding
+ * menu open when input containers are focused via click or tab.
+ */
+@Directive({
+  selector: '[md-menu-intput-item], [mdMenuInputItem], [mat-menu-intput-item], [matMenuInputItem]',
```
Comment (UltraMarco) [inline src/lib/menu/menu-input-item.ts]: input is misspelled

**Pull request closed without merging**